magicblock-committor-service/src/intent_executor/single_stage_executor.rs

```diff
@@ -0,0 +1,190 @@
+use std::ops::{ControlFlow, Deref};
+
+use log::{error, info};
+use magicblock_program::magic_scheduled_base_intent::ScheduledBaseIntent;
+
+use crate::{
+    intent_executor::{
+        error::{
+            IntentExecutorError, IntentExecutorResult,
+            TransactionStrategyExecutionError,
+        },
+        task_info_fetcher::TaskInfoFetcher,
+        ExecutionOutput, IntentExecutorImpl,
+    },
+    persist::IntentPersister,
+    tasks::task_strategist::TransactionStrategy,
+    transaction_preparator::TransactionPreparator,
+};
+
+pub struct SingleStageExecutor<'a, T, F> {
+    inner: &'a IntentExecutorImpl<T, F>,
+    // TODO: add strategy here?
+}
+
+impl<'a, T, F> SingleStageExecutor<'a, T, F>
+where
+    T: TransactionPreparator,
+    F: TaskInfoFetcher,
+{
+    pub fn new(executor: &'a IntentExecutorImpl<T, F>) -> Self {
+        Self { inner: executor }
+    }
+
+    pub async fn execute<P: IntentPersister>(
+        &self,
+        base_intent: ScheduledBaseIntent,
+        mut transaction_strategy: TransactionStrategy,
+        junk: &mut Vec<TransactionStrategy>,
```

Comment:
What is _junk_ here.
Are we throwing it away, retrying later, cleaning up later?

Again a more specific name would help to understand what kind of _junk_ we're looking at.